library/spdm_responder_lib/libspdm_rsp_receive_send.c

```diff
@@ -621,9 +621,11 @@ libspdm_return_t libspdm_build_response(void *context, const uint32_t *session_i
     #if LIBSPDM_ENABLE_CAPABILITY_CHUNK_CAP || LIBSPDM_ENABLE_CHUNK_CAP
     switch (request_response_code) {
     case SPDM_CHUNK_SEND_ACK:
-        request_response_code =
-            ((spdm_message_header_t*)(my_response + sizeof(spdm_chunk_send_ack_response_t)))
-            ->request_response_code;
+        if (my_response_size > sizeof(spdm_chunk_send_ack_response_t)) {
+            request_response_code =
+                ((spdm_message_header_t*)(my_response + sizeof(spdm_chunk_send_ack_response_t)))
+                ->request_response_code;
+        }
```

Comment:
In which scenario, this check will fail?

Reply:
When Chunk is not completed (send_info->chunk_bytes_transferred < send_info->large_message_size), the chunk_send response only contains the data of spdm_chunk_send_ack_response_t like (12 05 00 0f 02 00 ) which indicates chunk_seq_no.
When Chun is completed, the chunk_send response is like (12 05 00 0e 33 00 12 6e 00 00 ). So need to check the response size to get the response code in Chunk